src/main/java/de/danielbechler/diff/selector/CollectionItemElementSelector.java

```diff
@@ -71,6 +94,6 @@ public boolean equals(final Object o)
 	@Override
 	public int hashCode()
 	{
-		return item != null ? item.hashCode() : 0;
+		return item != null ? identityStrategy.hashCode(item) : 0;
```

Comment:
But, in Java, one never changes the behaviour of equals without changing the hashCode consistently. :) I think you are trying to protect the user from its own mistakes a bit too much.
Anyway, I have no problem with making it a constant.